### PR TITLE
Device hub server - JS engine - geo location passed using configuration

### PR DESCRIPTION
**Pull request by chespinoza**
This PR addresses #76 but instead of implementing a get_location() method, implements a pretty basic "goesque" mimic of Geolocation JS object.

```
JS Object: geolocation
    Integer Property: Timestamp
    JS Object: Coords
        Float Property: Latitude
        Float Property: Longitude
        Float Property: Accuracy
        Float Property: AltitudeAccuracy
        Float Property: Heading
        Float Property: Speed
```
In order to get latitude and longitude values configured on the server params:

```
geolocation.Coords.Latitude
geolocation.Coords.Longitude
```

The server's yaml config file now have the next new items:

```
geo_enabled: true
geo_lat: 54.416333
geo_lng: -4.36552
```

Also, JS engine has now a way to add new objects and functions to the JS runtime using maps

- AuxObjs for new Objects
- AuxFuncs for new functions

e.g Add a new Object:
```
	e := New(utils.NewNoOpLogger())

	pos := new(position)
	pos.Coords.Latitude = 54.416333
	pos.Coords.Longitude = -4.36552

	e.AuxObjs["geolocation"] = pos
```

### CODE REVIEW

File: examples/config/profile_script_geo.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  # Format of the message received.
+  # One of 'json', 'csv', 'xml', 'raw'
+  script-input: json
+  # Function to run - written in Javascript
+  # Can log via console.log
+  # Must return an object
+  script-contents: >
+      function xxx ( input )
+      {
+        return Number(geolocation.Coords.Latitude).toString() + ',' +  Number(geolocation.Coords.Longitude).toString()
```
Comment (mdevilliers) [inline examples/config/profile_script_geo.yaml]: Minor - Can the example be cleaned up to not include the casting as it should be a number already?

File: runtime/loop.go [hunk 2]
```diff
@@ -19,9 +19,14 @@ func loop(ctx context.Context,
 	endpoints map[string]hub.Endpoint,
 	channel hub.Channel,
 	logger utils.Logger,
-	tags map[string]string) {
+	tags map[string]string,
+	options map[string]interface{}) {
```
Comment (mdevilliers) [inline runtime/loop.go]: Could options be a struct rather than a map[string] interface?

File: engine/geolocation.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"time"
+)
+
+type coordinates struct {
+	Latitude         float64 // specified in decimal degrees
+	Longitude        float64 // specified in decimal degrees
+	Accuracy         float64 // specified in meters
+	Altitude         float64 // specified in meters
```
Comment (mdevilliers) [inline engine/geolocation.go]: Out of interest what happens if you try to access one of these properties - is it just undefined?

Comment (chespinoza) [inline engine/geolocation.go]: the value will be 0.

**Pull request merged**